### ISSUE TITLE
Mismatch of fields, common for vCard format?

### ISSUE THREAD

**Issue by JeroenSteen** [steven-elliott/vcard-json#6]
I tried to export it to JSON, and viewed it with jsonviewer.stack.hu.
I think the format is verry clear, better then of some other library I tried.

Only I saw some weird facts, sometimes:
zip is street or hoursenumber, street is one letter, country is city ect.
Is that common for vCard, how can I fix that?

**Comment by steven-elliott** [comment]
Haven't seen that before, but haven't tested all formats. Do you have the vcard version # or the vcard you are testing with strange results?

**Comment by JeroenSteen** [comment]
I used a export from GMail contacts, it was vCard version 3.0.
I don't know if it occured at this contact, but something like this:

```
BEGIN:VCARD
VERSION:3.0
FN:Anon
ADR;TYPE=HOME:;Ampeldam;Van Roonstraat 555A;Nederland;;230346 AV;
ADR;TYPE=HOME:;;Van Roonstraat 555A;Ampeldam;;23046 AV;Polen
END:VCARD
```

**Comment by steven-elliott** [comment]
Fixed in 0.5.2. All the fields for address are correct now. The error was related to the extended address in your example. I've ensured all fields parse correctly whether present or not.
